### PR TITLE
Allow capture of name during adding of card

### PR DESCRIPTION
**Pull request by mayeaux**
Screen Shot 2022-06-09 at 1 11 35 PM

### CODE REVIEW

File: ui/page/settingsStripeCard/view.jsx [hunk 1]
```diff
@@ -250,14 +257,22 @@ class SettingsStripeCard extends React.Component<Props, State> {
           });
 
           card.on('ready', function () {
-            card.focus();
+            // focus on the card input
```
Comment (tzarebczan) [inline ui/page/settingsStripeCard/view.jsx:260]: This comment is unnecessary :)

File: ui/page/settingsStripeCard/view.jsx [hunk 2]
```diff
@@ -353,11 +373,14 @@ class SettingsStripeCard extends React.Component<Props, State> {
...
               };
 
+              const formattedEmailName = cardDetails.topOfDisplay + ' ' + cardDetails.bottomOfDisplay;
+
               that.setState({
                 currentFlowStage: 'cardConfirmed',
                 pageTitle: 'Payment Methods',
                 userCardDetails: cardDetails,
                 paymentMethodId: customerStatusResponse.PaymentMethods[0].id,
+                cardName: customerStatusResponse.PaymentMethods[0].billing_details.name || formattedEmailName,
```
[the remark this answers is not in the document]
Comment (mayeaux) [reply ui/page/settingsStripeCard/view.jsx:383]: Yup it will

File: ui/scss/component/_stripe-card.scss [hunk 3]
```diff
@@ -423,3 +423,18 @@ pre {
 .currency-to-use-selector {
   margin: 0 auto;
 }
+
+.getting-card-status__div .card__header--between {
+  justify-content: center;
```
Comment (infinite-persistence) [inline ui/scss/component/_stripe-card.scss:428]: Why not just left-justify and truncate?  It looks like the max is 48 characters anyway, and the legacy email text can just be truncated to ellipsis.  When it wraps to 3 lines, the text is horizontally cut off (not shown here).

image

<sub>Hubert Blaine Wolfeschlegelsteinhausenbergerdorff Sr.</sub>
image

image

File: ui/page/settingsStripeCard/view.jsx [hunk 4]
```diff
@@ -395,9 +416,29 @@ class SettingsStripeCard extends React.Component<Props, State> {
       setPreferredCurrency(value);
     }
 
+    function onChangeCardName(event) {
+      const { value } = event.target;
+
+      const numberOrSpecialCharacter = /[0-9`!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?~]/;
```
Comment (infinite-persistence) [inline ui/page/settingsStripeCard/view.jsx:422]: - [ ] Is this overlimiting?  Was expecting the following to be valid names
  - Gopal s/o Raju
  - Hubert Lord Sr.
  - James Wolfe-Blain
- [ ] Add character count limit like an actual card, then we don't need unnecessary css.

image

image

File: ui/page/settingsStripeCard/view.jsx [hunk 5]
```diff
@@ -419,7 +460,11 @@ class SettingsStripeCard extends React.Component<Props, State> {
         {currentFlowStage === 'confirmingCard' && (
           <div className="sr-root">
             <div className="sr-main">
-              <div className="sr-payment-form card cardInput">
+              <div className="">
+                <div className="sr-form-row">
+                  <label className="payment-details">Name On Card</label>
```
Comment (infinite-persistence) [inline ui/page/settingsStripeCard/view.jsx:465]: i18n...

And would prefer `Name on card` for field labels, as that's what we've been trying to clean up everywhere, but it keeps getting back in.  

image

Capitalized version will only be for section titles, and occasionally buttons (although lower case is still preferred).

File: ui/page/settingsStripeCard/view.jsx [hunk 6]
```diff
@@ -410,7 +451,7 @@ class SettingsStripeCard extends React.Component<Props, State> {
 
         {/* initial markup to show while getting information */}
         {currentFlowStage === 'loading' && (
-          <div className="headerCard toConfirmCard">
+          <div className="headerCard getting-card-status__div">
```
Comment (infinite-persistence) [inline ui/page/settingsStripeCard/view.jsx:454]: No need to change, but hope to see less of it ... seen it a few times:

Having `div` as a BEM _element_ name is weird -- seen this in other PRs.  To the reader, it means "all `<div>`s under the `getting-card-status` block".

If the page has its own `stripe-card` class, then this should just be `stripe-card__getting-card-status`

File: ui/page/settingsStripeCard/view.jsx [hunk 7]
```diff
@@ -250,14 +257,20 @@ class SettingsStripeCard extends React.Component<Props, State> {
...
           });
 
           var email = that.props.email;
 
           function submitForm(event) {
             event.preventDefault();
 
+            const cardUserName = document.querySelector('#card-name').value;
+            if (!cardUserName) {
+              return (document.querySelector('.sr-field-error').innerHTML = __('Please enter the name on the card'));
```
Comment (infinite-persistence) [inline ui/page/settingsStripeCard/view.jsx:271]: error text should be the same color as `--color-text-error`

**Pull request merged**